optimization of history-bookmark.lua

dyphire opened the pull request:
- add options of special_protocols, it is used to excluded special protocols.
- add options of excluded_dir, it is used to excluded directories for cloud mount disks on Windows.

yuukidach commented:
@dyphire Thanks for creating a new PR. 

But I'm kind of busy these days. I'll review the new commits later.

dyphire committed:
optimization of history-bookmark.lua

add options of special_protocols, it is used to excluded special protocols.
add options of excluded_dir, it is used to excluded directories for cloud mount disks on Windows. example: ["X:", "Z:"]. !!the option only for Windows

yuukidach commented:
LGTM.  Sorry for reviewing so late.

Thanks for the contribution! 👍
